docs/csharp/nullable-references.md

```diff
@@ -92,46 +204,13 @@ You can also use directives to set these same contexts anywhere in your project:
-
-The compiler uses the following rules in a disabled nullable annotation context:
-
-- You can't declare nullable references in a disabled context.
-- All reference variables may be assigned a value of null.
-- No warnings are generated when a variable of a reference type is dereferenced.
-- The null-forgiving operator may not be used in a disabled context.
-
-The behavior is the same as previous versions of C#.
-
-The compiler uses the following rules in an enabled nullable annotation context:
-
-- Any variable of a reference type is a **non-nullable reference**.
-- Any non-nullable reference may be dereferenced safely.
-- Any nullable reference type (noted by `?` after the type in the variable declaration) may be null. Static analysis determines if the value is known to be non-null when it's dereferenced. If not, the compiler warns you.
-- You can use the null-forgiving operator to declare that a nullable reference isn't null.
-
-In an enabled nullable annotation context, the `?` character appended to a reference type declares a **nullable reference type**. The **null-forgiving operator** `!` may be appended to an expression to declare that the expression isn't null.
-
-## Nullable warning context
-
-The nullable warning context is distinct from the nullable annotation context. Warnings can be enabled even when the new annotations are disabled. The compiler uses static flow analysis to determine the **null state** of any reference. The null state is either **not null** or **maybe null** when the *nullable warning context* isn't **disabled**. If you dereference a reference when the compiler has determined it's **maybe null**, the compiler warns you. The state of a reference is **maybe null** unless the compiler can determine one of two conditions:
-
-1. The variable has been definitely assigned a non-null value.
-1. The variable or expression has been checked against null before de-referencing it.
-
-The compiler generates warnings when you dereference a variable or expression that is **maybe null** in a nullable warning context. Furthermore, the compiler generates warnings when a nonnullable reference-type variable is assigned a **maybe null** variable or expression in an enabled nullable annotation context.
-
-## Attributes describe APIs
-
-You add attributes to APIs that provide the compiler more information about when arguments or return values can or can't be null. You can learn more about these attributes in our article in the language reference covering the [nullable attributes](language-reference/attributes/nullable-analysis.md). These attributes are being added to .NET libraries over current and upcoming releases. The most commonly used APIs are being updated first.
-
 ## Known pitfalls
 
-Arrays and structs that contain reference types are known pitfalls in nullable reference types feature.
+Arrays and structs that contain reference types are known pitfalls in nullable references and the static analysis that determines null safety.
```

Comment:
How are the known pitfalls, what about them specifically? Like if they contain reference types, and one of the elements is `null`?

Reply:
Added an introductory statement here.